### PR TITLE
Improve Plan Dispatcher for Bridge - I

### PR DESCRIPTION
**Pull request by franklinselva**
The idea was to improve the plan dispatcher with respect to the execution context implemented in #27. Since the changes are significant, I would like to break the work into three stages.

1. The first stage extends the executable graph to dedicated Action Executors and the Plan dispatcher.
2. Solid integration of Plan Monitor with Plan Dispatcher for better error handling.
3. Integrate parallel execution with time-triggered actions and parallel actions.

This PR addresses the first stage of the work. Here, I have integrated dedicated action executors for discrete action types from UP and the plan dispatcher to handle the action executors. This PR also provides a baseline for the next two stages for error handling and parallel executions.

Note: Currently the plan dispatcher may not allow replanning within `PlanDispatcher` even though it can be triggered in the application directly.

### CODE REVIEW

**Comment by franklinselva** [comment]
@alexander-sung @sebastianstock This pull request should be now ready for review.

**Review by alexander-sung — APPROVED** [review]
The problem with the `parallel.py` example is not yet resolved in this pull request - are you aware?

All other examples and pytests run fine.

**Comment by franklinselva** [comment]
Yes. It should be fixed in the third segment of the actual improvement.

**Comment by franklinselva** [comment]
@sebastianstock Do you see any changes that need to be added? Could you please review this PR when you have a moment?

File: up_esb/execution/executor.py [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+            self._condition.notify()
+
+    def wait(self):
+        """Wait until all tasks are finished."""
+        with self._lock:
+            while self._tasks > 0:
+                self._condition.wait()
+
+
+class TaskExecutor:
```
Comment (sebastianstock) [inline up_esb/execution/executor.py:48]: It seems to me, that the `TaskExecutor` is not beeing used. Could be remove it or do you plan to use it in future developments?

File: up_esb/execution/executor.py [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+        self._dry_run = options.get("dry_run", False)
+        self._verbose = options.get("verbose", False)
+        self._context = None
+        self._action = None
+        self._node_name = None
+        self._parameters = None
+
+        self._supported_plan_kind = None
+
+    def get_executor(self, plan: Plan):
```
Comment (sebastianstock) [inline up_esb/execution/executor.py:75]: Here, I find it confusing in the examples that  `get_executor` is a method. So in the examples and tests you first create an `ActionExecutor` which is then updated to the subclass by calling this method.

**Review by sebastianstock — APPROVED** [review]
Hi  @franklinselva. Sorry for the delay and thanks for the update. I only have the two comments (see above) regarding the Executor. Both are not critical. Overall it looks good to me.

File: up_esb/execution/executor.py (hunk 1 above)
Comment (franklinselva) [reply up_esb/execution/executor.py:48]: I plan to use it to manage time-triggered executions and parallel execution.

**Pull request merged**